### ISSUE TITLE
sbt-sonatype no longer supports Java 8

### ISSUE THREAD

**Issue by armanbilge** [xerial/sbt-sonatype#530]
due to the Airframe update

- https://github.com/wvlet/airframe/pull/3648#issuecomment-2406249105

**Comment by xerial** [comment]
Thanks for identify the cause. As sbt-sonatype doesn't require logback-core, it should be excluded from the transitive dependencies

**Comment by xerial** [comment]
Released https://github.com/xerial/sbt-sonatype/releases/tag/v3.12.1

**Comment by xuwei-k** [comment]
does not work 3.12.1

```
java.lang.NoClassDefFoundError: ch/qos/logback/core/Context
	at wvlet.airframe.http.internal.LogRotationHttpLogger.<init>(LogRotationHttpLogger.scala:39)
	at wvlet.airframe.http.Compat$.$anonfun$defaultHttpClientLoggerFactory$1(Compat.scala:58)
	at wvlet.airframe.http.client.HttpClientConfig.newHttpLogger(HttpClientConfig.scala:178)
	at wvlet.airframe.http.client.SyncClient.$init$(SyncClient.scala:30)
	at wvlet.airframe.http.client.SyncClientImpl.<init>(SyncClient.scala:155)
	at wvlet.airframe.http.client.HttpClientBackend.newSyncClient(HttpClientBackend.scala:28)
	at wvlet.airframe.http.client.HttpClientBackend.newSyncClient$(HttpClientBackend.scala:27)
	at wvlet.airframe.http.client.URLConnectionClientBackend$.newSyncClient(URLConnectionClientBackend.scala:20)
	at wvlet.airframe.http.client.HttpClientConfig.newSyncClient(HttpClientConfig.scala:62)
	at xerial.sbt.sonatype.SonatypeClient.<init>(SonatypeClient.scala:67)
	at xerial.sbt.Sonatype$.withSonatypeService(Sonatype.scala:519)
	at xerial.sbt.Sonatype$.$anonfun$sonatypeBundleRelease$1(Sonatype.scala:241)
```

**Comment by xerial** [comment]
I forgot about the logger usage for http clients. Logging might needs to be disabled

**Comment by xerial** [comment]
Released a hotfix version https://github.com/xerial/sbt-sonatype/releases/tag/v3.12.2

**Comment by xerial** [comment]
Note: Java 8 is no longer supported. Consider using Java 17 (LTS) or later

**Comment by eed3si9n** [comment]
Hello. Would you reconsider this decision to drop JDK 8 support plz?

There are still number of tooling such as Scala 2.x, Scala 3.3 LTS, and sbt that are built and published on JDK 8, and given that Sonatype has announced [the sunset of Sonatype OSSRH](https://central.sonatype.org/news/20250326_ossrh_sunset/) (OSS Repository Hosting), we'd be using sbt-sonatype or some other mechanism to publish to Central Publisher Portal to eventually publish to Maven Central. /cc @SethTisue, @tgodzik

**Comment by xerial** [comment]
@eed3si9n Thanks for the heads up. Internal HTTP client airframe-http (for legacy Sonatype OSSRH) still works for Java8. We need to revisit the current implementation of a new central client (based on sttp) to ensure it will continue to support Java8. 

Anyway, as OSSRH will sunset by June 2025, I'll take a look by then.

**Comment by Andrapyre** [comment]
@eed3si9n , I'm assuming Java 8 support is a non-negotiable and that you can't just announce Java 11 as a minimum version? Basically all of the widely used http clients (like sttp, requests, etc.) have Java 11 as minimum support. If that's a must, I can offer to extend the sonatype central client under two scenarios (unless someone has a better suggestion):

1. A new Java 8 compat lib based on Airframe, since that's what sbt-sonatype prefers. I would need @xerial  or someone else to first enable support for multipart uploads, as described in https://github.com/wvlet/airframe/issues/3420

2. A new Java 8 compat lib based on`HttpUrlConnection`. 

Either way, it would be great to have a reviewer for this, as most of my Java xp is Java 11+.

**Comment by eed3si9n** [comment]
I wrote a HTTP library on JDK 8 over the weekend - https://eed3si9n.com/gigahorse-0.8.0 . I don't think it can handle multi-part upload yet, but hopefully it won't be too bad to add support for it. It's a matter of constructing special kind of text out of files and key-values, I think.

**Comment by Andrapyre** [comment]
Super - I've created https://github.com/lumidion/sonatype-central-client/issues/16 to track this. Just drop a note when you have multipart upload support and I can take the wheel to create a new sonatype central client lib.

**Comment by eed3si9n** [comment]
Released Gigahorse 0.9.0 with multipart support - https://eed3si9n.com/gigahorse-0.9.0

**Comment by Andrapyre** [comment]
@eed3si9n , is the gigahorse sonatype central client I've created good for going live? I saw you closed a PR for changing the gigahorse library.
